lib/ramble/ramble/spack_runner.py

```diff
@@ -345,6 +351,28 @@ def get_package_path(self, package_spec):
             self._dry_run_print(args)
             return '/dry-run/path/to/%s' % package_spec.split()[0]
 
+    def mirror_environment(self, mirror_path):
+        """Create a spack mirror from the activated environment"""
+        self._check_active()
+
+        args = [
+            "mirror",
+            "create",
+            "--all",  # All packages in the environment
+            "-D",  # Include dependencies
+            "-d",
+            mirror_path
+        ]
+
+        if not self.dry_run:
+            return self.exe(*args, output=str).strip()
+        else:
+            self._dry_run_print(args)
+            return """
+0    already present
```

Comment:
If we're going not DRY / repeat this print perhaps we should use the same logic in the places it repeats so we're less likely to miss it in a future cleanup? 

It is a bit of a pain to DRY though..
```
                "  %-4d already present"  % len(self._software_mirror_stats.present),
                "  %-4d added"            % len(self._software_mirror_stats.new),
                "  %-4d failed to fetch." % len(self._software_mirror_stats.errors))
```

Reply:
I updated the logic, but the spack runner can't reference `self._software_mirror_stats`. That's only a member of the workspace. Here we're just passing the string that spack would have printed (if we called it directly) to give the stats of the software mirror creation process.

Ideally, this would call spack methods directly, instead of calling spack as a sub-process. But for now we still have spack as an external dep.

Reply:
Makes sense, thanks!